Documentation/upgrade.md

```diff
@@ -0,0 +1,293 @@
+---
+title: Upgrading Rook
+weight: 40
+---
+
+# Upgrading the Rook Software
+This guide will walk you through the manual steps to upgrade the software in a Rook cluster from one version to the next.
+Rook is a distributed software system and therefore there are multiple components to individually upgrade in the sequence defined in this guide.
+After each component is upgraded, it is important to verify that the cluster returns to a healthy and fully functional state.
+
+
+This guide is just the beginning of upgrade support in Rook.
+The goal is to provide prescriptive guidance and knowledge on how to upgrade a live Rook cluster and we hope to get valuable feedback from the community that will be incorporated into an automated upgrade solution by the Rook operator.
+
+We welcome feedback and opening issues!
+
+## Considerations
+With this manual upgrade guide, there are a few notes to consider:
+* **WARNING:** Upgrading a Rook cluster is a manual process in its very early stages.  There may be unexpected issues or obstacles that damage the integrity and health of your storage cluster, including data loss.  Only proceed with this guide if you are comfortable with that.
+* Rook is still in an alpha state.  Migrations and general support for breaking changes across versions are not supported or covered in this guide.
+
+## Prerequisites
+In order to successfully upgrade a Rook cluster, the following prerequisites must be met:
+* The cluster should be in a healthy state with full functionality.
+Review the [health verification section](#health-verification) in order to verify your cluster is in a good starting state.
+* `dataDirHostPath` must be set in your Cluster spec.
+This persists metadata on host nodes, enabling pods to be terminated during the upgrade and for new pods to be created in their place.
+More details about `dataDirHostPath` can be found in the [Cluster CRD readme](./cluster-crd.md#cluster-settings).
+
+The minimal sample Cluster spec that will be used in this guide can be found below (note that the specific configuration may not be applicable to all environments):
```

Comment:
Another prerequisite i would add is to not have any in-flight request being done. Like a PVC being created or a user pod attaching a rbd volume.

Reply:
I can add this as a pre-req also